src/themes/DarkThemeProvider.tsx

```diff
@@ -1,14 +1,41 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useRef } from 'react';
 import useMediaQuery from '@material-ui/core/useMediaQuery';
 
 import DarkThemeContext from './DarkThemeContext';
 
 const DarkThemeProvider: React.FC = ({ children }) => {
   const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');
   const [darkMode, setDarkMode] = React.useState(prefersDarkMode);
+  const firstRender=useRef(true);
 
   useEffect(() => {
-    setDarkMode(prefersDarkMode);
+    const persistDarkMode=()=>localStorage.setItem('rediJobBoardIsDarkMode', darkMode? 'true' : 'false');
+    window.addEventListener('beforeunload', persistDarkMode); // before unloading the app store the dark mode. Questionable if we always want to persist the darkmode tho
+    
+    return ()=>{
+      window.removeEventListener('beforeunload', persistDarkMode);
+    };
+  }, [darkMode]);
+
+  useEffect(()=>{
+    if(firstRender.current){ // if this useEffect fires for the first time prio the darkMode value that is stored inside local storage
+      const isDark=localStorage.getItem('rediJobBoardIsDarkMode');
+      switch(isDark){ // can also json parse it before but seems overkill
+      case 'true':
+        setDarkMode(true);
+        break;
+      case 'false':
+        setDarkMode(false);
+        break;
+      default: 
+        setDarkMode(prefersDarkMode);
+      }
+      firstRender.current=false;
+    }else {
```

Comment:
```suggestion
    } else {
```
:P Need to check why the linter let this one pass!